code/__HELPERS/ray_helpers.dm

```diff
@@ -0,0 +1,586 @@
+/**
+  * Any unsorted global procs that help when locating turfs,  via
+  * casting rays out in specified directions etc
+  */
+
+/**
+  * Returns the turf offset from a given atom by the given dx and dy
+  * parameters
+  *
+  * This proc is bound to the map edges so will not overrun them
+  *
+  * Arguments:
+  * * atom/A the atom to calculate the offset from (the source)
+  * * dx, how many steps on the map grid to go in the x direction
+  * * dy, how many steps on the map grid to go in the y direction
+  */
+/proc/get_offset_target_turf(atom/A, dx, dy)
+	var/x = min(world.maxx, max(1, A.x + dx))
+	var/y = min(world.maxy, max(1, A.y + dy))
+	return locate(x,y,A.z)
+
+/**
+  * Get turfs on the edge of a bounding box
+  *
+  * Gets the turfs on the edge of a bounding box of
+  * given x and y height/width that surround the passed
+  * in source atom, uses [proc/get_offset_target_turf]
+  * internally so will be bounded by map edges
+  *
+  * Arguments:
+  * * atom/A the atom to place the box around
+  * * radius_x radius of the box in the x dimension
+  * * radius_y the radius of the box in the y dimension
+  */
+/proc/get_bounding_turfs(atom/A, radius_x, radius_y)
```

Comment:
please review this, it's the only new proc

Reply:
Why not combine identical for?
```suggestion
	//Go along the y axis from start x
	for(var/y = start_y; y < end_y; y++)
		turfs |= get_offset_target_turf(A, start_x, y)

	//Go along the y axis from end x
		turfs |= get_offset_target_turf(A, end_x, y)

	// Go along the x axis from start y
	for(var/x = start_x; x < end_x; x++)
		turfs |= get_offset_target_turf(A, x, start_y)

	// Go along the x axis from end y
		turfs |= get_offset_target_turf(A, x, end_y)
```

Reply:
This better than 2 `block`?
`
turfs = block(get_offset_target_turf(A, start_x, start_y),get_offset_target_turf(A, end_x, end_y))-block(get_offset_target_turf(A, start_x+1, start_y+1),get_offset_target_turf(A, end_x-1, end_y-1))
`

Reply:
I forgot we had block, I'll do this

Reply:
actually, I guess it would depend on the size of the radius which we choose as to which ends up faster at some point

Reply:
Now this double add corner turfs. i think, change in `for` along the x axis make it possible safe change `|=` to `+=`
```
	//Go along the y axis from start x
	for(var/y = start_y; y <= end_y; y++)
		turfs += get_offset_target_turf(A, start_x, y)

	//Go along the y axis from end x
		turfs += get_offset_target_turf(A, end_x, y)

	// Go along the x axis from start y
	for(var/x = start_x+1; x <= end_x-1; x++)
		turfs += get_offset_target_turf(A, x, start_y)

	// Go along the x axis from end y
		turfs += get_offset_target_turf(A, x, end_y)
```